### PR TITLE
NMS --classes 0 bug fix

### PR DESCRIPTION
**Pull request by gkagkos**
I found that if I use only 0 to filter the classes in the non_max_suppression then it won't go through the if statement for the classes. Therefore I added a quick fix.

```python
if classes is not None:
    x = x[(x[:, 5:6] == torch.tensor(classes, device=x.device)).any(1)]

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Enhancement of class filtering in non-maximum suppression function.

### 📊 Key Changes
- Updated the condition for class filtering from `if classes:` to `if classes is not None:`

### 🎯 Purpose & Impact
- **Purpose:** Ensure that class filtering is correctly applied only when `classes` variable is explicitly set, not just when it's truthy.
- **Impact:** Prevents potential bugs and makes behavior more predictable for end-users when specifying classes for object detection models. This should improve the model's accuracy in scenarios where selective class detection is needed. 🎯🐞

This change will not be immediately noticeable to end-users but ensures more reliable and precise object detection in development.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@polydefk ah,  yes, good idea and good job catching this bug!

**Comment by glenn-jocher** [comment]
@polydefk PR is merged! Thank you for your contributions.
